crates/engine/tree/src/tree/mod.rs

```diff
@@ -1209,27 +1209,27 @@ where
                 Err(TryRecvError::Closed) => return Err(TryRecvError::Closed.into()),
                 Err(TryRecvError::Empty) => self.persistence_state.rx = Some((rx, start_time)),
             }
+            return Ok(())
```

Comment:
I'm not sure about this @klkvr 

I believe we always want to try to queue in the next pending persistence task via

`self.persistence_state.remove_above_state.pop_front()`

so we can't return early here

ref https://github.com/paradigmxyz/reth/pull/13928